components/compass_theme_provider/compass_theme_provider.tsx

```diff
@@ -20,6 +87,135 @@ const CompassThemeProvider = ({theme, children}: Props): JSX.Element | null => {
+            --mention-highlight-bg-mixed-rgb: ${stripRGBADeclarator(dropAlpha(blendColors(theme.centerChannelBg, theme.mentionHighlightBg)))};
+            --pinned-highlight-bg-mixed-rgb: ${stripRGBADeclarator(dropAlpha(blendColors(theme.centerChannelBg, theme.mentionHighlightBg)))};
+            --own-highlight-bg-rgb: ${stripRGBADeclarator(dropAlpha(blendColors(theme.mentionHighlightBg, theme.centerChannelColor)))};
+            --collapsed-post-bg-mixed-rgb: ${stripRGBADeclarator(dropAlpha(blendColors(theme.centerChannelBg, theme.centerChannelColor)))};
+            --new-message-separator-rgb: ${stripRGBDeclarator(convertToRgb(theme.newMessageSeparator))};
+            --online-indicator-rgb: ${stripRGBDeclarator(convertToRgb(theme.onlineIndicator))};
+            --sidebar-bg-rgb: ${stripRGBDeclarator(convertToRgb(theme.sidebarBg))};
+            --sidebar-header-bg-rgb: ${stripRGBDeclarator(convertToRgb(theme.sidebarHeaderBg))};
+            --sidebar-teambar-bg-rgb: ${stripRGBDeclarator(convertToRgb(theme.sidebarTeamBarBg))};
+            --sidebar-header-text-color-rgb: ${stripRGBDeclarator(convertToRgb(theme.sidebarHeaderTextColor))};
+            --sidebar-text-rgb: ${stripRGBDeclarator(convertToRgb(theme.sidebarText))};
+            --sidebar-text-active-border-rgb: ${stripRGBDeclarator(convertToRgb(theme.sidebarTextActiveBorder))};
+            --sidebar-text-active-color-rgb: ${stripRGBDeclarator(convertToRgb(theme.sidebarTextActiveColor))};
+            --sidebar-text-hover-bg-rgb: ${stripRGBDeclarator(convertToRgb(theme.sidebarTextHoverBg))};
+            --sidebar-unread-text-rgb: ${stripRGBDeclarator(convertToRgb(theme.sidebarUnreadText))};
+
+            // Hex CSS variables
+            --sidebar-bg: ${theme.sidebarBg};
+            --sidebar-text: ${theme.sidebarText};
+            --sidebar-unread-text: ${theme.sidebarUnreadText};
+            --sidebar-text-hover-bg: ${theme.sidebarTextHoverBg};
+            --sidebar-text-active-border: ${theme.sidebarTextActiveBorder};
+            --sidebar-text-active-color: ${theme.sidebarTextActiveColor};
+            --sidebar-header-bg: ${theme.sidebarHeaderBg};
+            --sidebar-teambar-bg: ${theme.sidebarTeamBarBg};
+            --sidebar-header-text-color: ${theme.sidebarHeaderTextColor};
+            --online-indicator: ${theme.onlineIndicator};
+            --away-indicator: ${theme.awayIndicator};
+            --dnd-indicator: ${theme.dndIndicator};
+            --mention-bg: ${theme.mentionBg};
+            --mention-color: ${theme.mentionColor};
+            --center-channel-bg: ${theme.centerChannelBg};
+            --center-channel-color: ${theme.centerChannelColor};
+            --new-message-separator: ${theme.newMessageSeparator};
+            --link-color: ${theme.linkColor};
+            --button-bg: ${theme.buttonBg};
+            --button-bg-hover: ${rgbToHex(changeColor(theme.buttonBg, -0.15))};
```

Comment:
Can we pass the hex color here instead so we avoid having to convert? I am generally against having too many color conversion functions.

Reply:
the problem is, that the `buttonBg` color is being altered programmatically and the `changeColor` helper returns a rgb string ... so no (atm)